### PR TITLE
Added query by NSPredicate String capability.

### PR DESCRIPTION
**Pull request by rafael-chavez**

### CODE REVIEW

**Comment by rafael-chavez** [comment]
@marekcirkos

**Comment by ericwengelking** [comment]
Our hope is to get something like MobileBy.iOSNSPredicate() added to Appium once it supports XCUITest.  I've opened an issue here https://github.com/appium/appium-xcuitest-driver/issues/27

FYI @rafael-chavez

File: XCTWebDriverAgentLib/Commands/FBFindElementCommands.m [hunk 1]
```diff
@@ -206,6 +208,13 @@ + (void)descendantsOfElement:(XCUIElement *)element withProperty:(NSString *)pro
   [results addObjectsFromArray:childElements];
 }
 
+#pragma mark - Search by Predicate String
++ (NSArray *)descendantsOfElement:(XCUIElement *)element withPredicateString:(NSString *)predicateString {
+    XCUIElementQuery *query = [[element descendantsMatchingType:XCUIElementTypeAny] matchingPredicate:[NSPredicate predicateWithFormat:predicateString]];
```
Comment (marekcirkos) [inline XCTWebDriverAgentLib/Commands/FBFindElementCommands.m]: I guess it will not work with UIAutomation class names? Which we should remove anyway ;).

**Comment by marekcirkos** [comment]
@rafael-chavez Thanks for PR. Looks good to me. Please sign CLA & we can shit it :)

File: XCTWebDriverAgentLib/Commands/FBFindElementCommands.m (hunk 1 above)
Comment (rafael-chavez) [inline XCTWebDriverAgentLib/Commands/FBFindElementCommands.m]: Correct

**Comment by rafael-chavez** [comment]
@marekcirkos Already signed the CLA.

**Comment by marekcirkos** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**